File: MafiaPediaFront/src/types/clubPlay.ts
export interface ReplaceParticipantDto {
  newClubPlayerId: number
  isGuest: boolean
  entryCount?: number
}

export interface MasterContextDto {
  masterId: number
  masterName: string
  clubId: number
  clubName: string
}

export interface ParticipantInputDto {
  clubPlayerId: number
  isGuest: boolean
  entryCount?: number
}

export interface CreateClubPlayDto {
  title?: string
  dateTime: string
  roomId: number
  senarioId: number
  playersCount: number
  desc?: string
  link?: string
  playType: 'normal' | 'rank' | 'superrank' | 'etc'
  eventId?: number | null
  nerkhId?: number | null
  shuffleRoles?: boolean
  masterId?: number
  participants: ParticipantInputDto[]
}

export interface ClubPlayParticipantDto {
  id: number
  clubPlayerId: number
  name: string
  roleId: number
  roleName: string
  sideId: number
  rolePhoto: string | null
  isGuest: boolean
  entryCount: number
}

export interface ClubPlayDetailDto {
  id: number
  title: string | null
  dateTime: string
  roomId: number
  roomName: string
  senarioId: number
  senarioName: string
  playersCount: number
  guestCount: number
  desc: string | null
  link: string | null
  playType: string
  status: string
  masterId: number
  masterName: string
  winnersideId: number | null
  eventId: number
  eventName: string
  nerkhId: number | null
  nerkhName: string | null
  participants: ClubPlayParticipantDto[]
}

export interface ClubPlayListItemDto {
  id: number
  title: string | null
  dateTime: string
  businessDate: string
  roomName: string
  senarioName: string
  playersCount: number
  guestCount: number
  status: string
  playType: string
  masterName?: string | null
}

export interface ClubPlayDeletedListItemDto extends ClubPlayListItemDto {
  deletedAt: string | null
  deletedByUserId: number | null
  deletedByDisplayName: string | null
}

export interface MasterStatsDto {
  totalPlays: number
  totalEntries: number
  totalGuestEntries: number
}

export interface MasterPerformanceDto {
  masterId: number
  masterName: string
  playCount: number
  entryCount: number
  guestEntryCount: number
}

export interface EventDto {
  id: number
  name: string
  clubId: number
  isDefault: boolean
}

File: MafiaPedia.Api/Entities/Clubplay.cs
using System;
using System.Collections.Generic;

namespace MafiaPedia.Api.Entities;

public partial class Clubplay
{
    public int Id { get; set; }

    public string? Title { get; set; }

    public DateTime? DateTime { get; set; }

    public int? PlayersCount { get; set; }

    public int? GuestCount { get; set; }

    public string? Desc { get; set; }

    public int RoomId { get; set; }

    public int SenarioId { get; set; }

    public int MasterId { get; set; }

    public int? WinnersideId { get; set; }

    public int UserId { get; set; }

    public int EventId { get; set; }

    public string? Link { get; set; }

    public string PlayType { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateOnly? BusinessDate { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime? DeletedAt { get; set; }

    public int? DeletedByUserId { get; set; }

    public int? NerkhId { get; set; }

    public virtual ICollection<Clubplayplayer> Clubplayplayers { get; set; } = new List<Clubplayplayer>();

    public virtual User? DeletedByUser { get; set; }

    public virtual Event Event { get; set; } = null!;

    public virtual Master Master { get; set; } = null!;

    public virtual Nerkh? Nerkh { get; set; }

    public virtual Room Room { get; set; } = null!;

    public virtual Senario Senario { get; set; } = null!;

    public virtual User User { get; set; } = null!;

    public virtual Side? Winnerside { get; set; }
}
